src/compile/data/bin.ts

```diff
@@ -38,15 +38,18 @@ function getSignalsFromModel(model: Model, key: string) {
   };
 }
 
+// Bad Name?
 function isBinTransform(t: FieldDef<string> | BinTransform): t is BinTransform {
   return 'as' in t;
 }
 
 function createBinComponent(t: FieldDef<string> | BinTransform, model: Model) {
   let as: [string, string];
 
-  if (isBinTransform(t)) {
+  if (isBinTransform(t) && isString(t.as)) {
     as = [t.as, `${t.as}_end`];
+  } else if (isBinTransform(t)) {
```

Comment:
I think you better add `if (isString(t.as))` in side the original `isBinTransform` if branch